### PR TITLE
Typehinted query namespace

### PR DESCRIPTION
**Pull request by massimilianobraglia**
see #1548

### CODE REVIEW

**Comment by massimilianobraglia** [comment]
The problem here is in the `InnerHits::toArray` method.
```php
class InnerHits
{
    [...]
    /**
     * @return array
     */
    public function toArray()
    {
        $array = parent::toArray();

        // if there are no params, it's ok, but ES will throw exception if json
        // will be like {"top_hits":[]} instead of {"top_hits":{}}
        if (empty($array['inner_hits'])) {
            $array['inner_hits'] = new \stdClass();
        }

        return $array['inner_hits'];
    }
    [...]
}
```

The problem is that I cannot set `array` as the return type (this one returns a `new \stdClass` because of the comment).

And this causes that I cannot set it to the base `ArrayableInterface`.

I hope I explained it better 😄

File: lib/Elastica/Query/Common.php [hunk 1]
```diff
@@ -90,7 +90,7 @@ public function setLowFrequencyOperator($operator)
      *
      * @return $this
      */
-    public function setHighFrequencyOperator($operator)
+    public function setHighFrequencyOperator(string $operator): self
     {
         return $this->setQueryParam('high_frequency_operator', $operator);
     }
```
Comment (p365labs) [inline lib/Elastica/Query/Common.php]: - setLowFrequencyOperator
- setHighFrequencyOperator

```default value or```

File: lib/Elastica/Query/FunctionScore.php [hunk 2]
```diff
@@ -242,7 +260,7 @@ public function setMaxBoost($maxBoost)
      *
      * @return $this
      */
-    public function setBoostMode($mode)
+    public function setBoostMode(string $mode): self
```
Comment (p365labs) [inline lib/Elastica/Query/FunctionScore.php]: default value ```multiply ```

Comment (p365labs) [reply lib/Elastica/Query/FunctionScore.php]: - boost_mode
- score_mode

File: lib/Elastica/Query/GeoDistance.php [hunk 3]
```diff
@@ -48,7 +48,7 @@ public function setDistance($distance)
      *
      * @return $this
      */
-    public function setDistanceType($distanceType)
+    public function setDistanceType(string $distanceType): self
```
Comment (p365labs) [inline lib/Elastica/Query/GeoDistance.php]: default value ```arc```

File: lib/Elastica/Query/InnerHits.php [hunk 4]
```diff
@@ -49,9 +49,9 @@ public function setName($name)
      *
      * @return $this
      */
-    public function setSize($size)
+    public function setSize(int $size): self
```
Comment (p365labs) [inline lib/Elastica/Query/InnerHits.php]: default vale for size

File: lib/Elastica/Query/Match.php [hunk 5]
```diff
@@ -194,7 +194,7 @@ public function setFieldMaxExpansions($field, $maxExpansions)
      *
      * @return $this
      */
-    public function setFieldZeroTermsQuery($field, $zeroTermQuery = self::ZERO_TERM_NONE)
+    public function setFieldZeroTermsQuery(string $field, string $zeroTermQuery = self::ZERO_TERM_NONE): self
```
Comment (p365labs) [inline lib/Elastica/Query/Match.php]: default value ```zero_terms_query```

File: lib/Elastica/Query/MoreLikeThis.php [hunk 6]
```diff
@@ -43,9 +41,9 @@ public function setLike($like)
      *
      * @return $this
      */
-    public function setBoost($boost)
+    public function setBoost(float $boost): self
```
Comment (p365labs) [inline lib/Elastica/Query/MoreLikeThis.php]: default value to ```1.0```

File: lib/Elastica/Query/MoreLikeThis.php [hunk 7]
```diff
@@ -55,9 +53,9 @@ public function setBoost($boost)
      *
      * @return $this
      */
-    public function setMaxQueryTerms($maxQueryTerms)
+    public function setMaxQueryTerms(int $maxQueryTerms): self
```
Comment (p365labs) [inline lib/Elastica/Query/MoreLikeThis.php]: default value for ```max_query_terms```

File: lib/Elastica/Query/MoreLikeThis.php [hunk 8]
```diff
@@ -115,9 +113,9 @@ public function setMinWordLength($minWordLength)
      *
      * @return $this
      */
-    public function setMaxWordLength($maxWordLength)
+    public function setMaxWordLength(int $maxWordLength): self
     {
-        return $this->setParam('max_word_length', (int) $maxWordLength);
+        return $this->setParam('max_word_length', $maxWordLength);
     }
```
Comment (p365labs) [inline lib/Elastica/Query/MoreLikeThis.php]: - min_term_freq
- min_doc_freq
- max_doc_freq
- min_word_length
etc.. 

have default values

Comment (p365labs) [reply lib/Elastica/Query/MoreLikeThis.php]: also 
- boost_terms

File: lib/Elastica/Query/Nested.php [hunk 9]
```diff
@@ -41,7 +41,7 @@ public function setQuery(AbstractQuery $query)
      *
      * @return $this
      */
-    public function setScoreMode($scoreMode)
+    public function setScoreMode(string $scoreMode): self
```
Comment (p365labs) [inline lib/Elastica/Query/Nested.php]: default to ```avg```

File: lib/Elastica/Query/ParentId.php [hunk 10]
```diff
@@ -41,7 +41,7 @@ private function setId($id)
     /**
      * @param bool $ignoreUnmapped
      */
-    private function setIgnoreUnmapped($ignoreUnmapped)
+    private function setIgnoreUnmapped(bool $ignoreUnmapped)
```
Comment (p365labs) [inline lib/Elastica/Query/ParentId.php]: defaults to ```false```

File: lib/Elastica/Query/QueryString.php [hunk 11]
```diff
@@ -71,7 +65,7 @@ public function setDefaultField($field)
      *
      * @return $this
      */
-    public function setDefaultOperator($operator)
+    public function setDefaultOperator(string $operator): self
```
Comment (p365labs) [inline lib/Elastica/Query/QueryString.php]: defaul ```or``` operator

File: lib/Elastica/Query/SimpleQueryString.php [hunk 12]
```diff
@@ -52,7 +52,7 @@ public function setFields(array $fields)
      *
      * @return $this
      */
-    public function setDefaultOperator($operator)
+    public function setDefaultOperator(string $operator): self
```
Comment (p365labs) [inline lib/Elastica/Query/SimpleQueryString.php]: ```or``` is the default operator

**Comment by massimilianobraglia** [comment]
@p365labs inserted the default values! Can you give it a look?

**Pull request merged**